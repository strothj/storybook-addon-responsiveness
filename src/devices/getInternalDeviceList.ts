import * as lzstring from "lz-string";
import DeviceList from "./DeviceList";
import compressedInternalDeviceList from "./compressedInternalDeviceList";

const getInternalDeviceList = (): DeviceList =>
  JSON.parse(
    lzstring.decompressFromEncodedURIComponent(compressedInternalDeviceList),
  ) as DeviceList;

export default getInternalDeviceList;
