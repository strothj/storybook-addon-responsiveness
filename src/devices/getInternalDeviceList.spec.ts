import getInternalDeviceList from "./getInternalDeviceList";

it("returns device list from module", () => {
  const devices = getInternalDeviceList();
  const iPhone5 = devices.extensions.find(d =>
    d.device.title.includes("iPhone 5"),
  );

  expect(iPhone5).toMatchSnapshot();
});
