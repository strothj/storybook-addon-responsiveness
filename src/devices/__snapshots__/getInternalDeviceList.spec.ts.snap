// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns device list from module 1`] = `
Object {
  "device": Object {
    "capabilities": Array [
      "touch",
      "mobile",
    ],
    "modes": Array [
      Object {
        "insets": Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
        "orientation": "vertical",
        "title": "default",
      },
      Object {
        "insets": Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
        "orientation": "horizontal",
        "title": "default",
      },
    ],
    "screen": Object {
      "device-pixel-ratio": 2,
      "horizontal": Object {
        "height": 320,
        "outline": Object {
          "image": "@url(iPhone5-landscape.svg)",
          "insets": Object {
            "bottom": 28,
            "left": 115,
            "right": 115,
            "top": 25,
          },
        },
        "width": 568,
      },
      "vertical": Object {
        "height": 568,
        "outline": Object {
          "image": "@url(iPhone5-portrait.svg)",
          "insets": Object {
            "bottom": 111,
            "left": 29,
            "right": 25,
            "top": 105,
          },
        },
        "width": 320,
      },
    },
    "show-by-default": true,
    "title": "iPhone 5/SE",
    "type": "phone",
    "user-agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.0 Mobile/14E304 Safari/602.1",
  },
  "order": 30,
  "type": "emulated-device",
}
`;
